spring-cloud-starter-single-step-batch-job/src/test/java/org/springframework/cloud/task/batch/autoconfigure/FlatFileItemWriterAutoConfigurationTests.java

```diff
@@ -0,0 +1,532 @@
+
+		applicationContextRunner.run((context) -> {
+			JobLauncher jobLauncher = context.getBean(JobLauncher.class);
+
+			Job job = context.getBean(Job.class);
+
+			JobExecution jobExecution = jobLauncher.run(job, new JobParameters());
+
+			JobExplorer jobExplorer = context.getBean(JobExplorer.class);
+
+			while (jobExplorer.getJobExecution(jobExecution.getJobId()).isRunning()) {
+				Thread.sleep(1000);
+			}
+
+			AssertFile.assertLineCount(2, this.outputFile);
+
+			String results = FileCopyUtils.copyToString(new InputStreamReader(
+					new FileSystemResource(this.outputFile).getInputStream()));
+			assertThat(results).isEqualTo("item = foo\nitem = bar\n");
+		});
+	}
+
+	@Test
+	public void testFormattedFieldExtractorFileGeneration() {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(FormattedFieldExtractorJobConfiguration.class)
+				.withConfiguration(
+						AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+								BatchAutoConfiguration.class,
+								SingleStepJobAutoConfiguration.class,
+								FlatFileItemWriterAutoConfiguration.class))
+				.withPropertyValues("spring.batch.job.jobName=job",
+						"spring.batch.job.stepName=step1", "spring.batch.job.chunkSize=2",
+						"spring.batch.job.flatfilewriter.name=fooWriter",
+						String.format(
+								"spring.batch.job.flatfilewriter.resource=file://%s",
+								this.outputFile.getAbsolutePath()),
+						"spring.batch.job.flatfilewriter.encoding=UTF-8",
+						"spring.batch.job.flatfilewriter.formatted=true",
+						"spring.batch.job.flatfilewriter.names=item",
+						"spring.batch.job.flatfilewriter.format=item = %s");
+
+		applicationContextRunner.run((context) -> {
+			JobLauncher jobLauncher = context.getBean(JobLauncher.class);
+
+			Job job = context.getBean(Job.class);
+
+			JobExecution jobExecution = jobLauncher.run(job, new JobParameters());
+
+			JobExplorer jobExplorer = context.getBean(JobExplorer.class);
+
+			while (jobExplorer.getJobExecution(jobExecution.getJobId()).isRunning()) {
+				Thread.sleep(1000);
+			}
+
+			AssertFile.assertLineCount(3, this.outputFile);
+
+			String results = FileCopyUtils.copyToString(new InputStreamReader(
+					new FileSystemResource(this.outputFile).getInputStream()));
+			assertThat(results).isEqualTo("item = f\nitem = b\nitem = b\n");
+		});
+	}
+
+	@Test
+	public void testFieldExtractorFileGeneration() {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(FieldExtractorConfiguration.class)
+				.withConfiguration(
+						AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+								BatchAutoConfiguration.class,
+								SingleStepJobAutoConfiguration.class,
+								FlatFileItemWriterAutoConfiguration.class))
+				.withPropertyValues("spring.batch.job.jobName=job",
+						"spring.batch.job.stepName=step1", "spring.batch.job.chunkSize=5",
```

Comment:
Just curious.  Why does each test has a different chunk size.

Reply:
I fixed it so they all use the same except the one that needs to be different.